validator/sawtooth_validator/journal/chain.py

```diff
@@ -473,3 +479,38 @@ def on_block_received(self, block):
         # pylint: disable=broad-except
         except Exception as exc:
             LOGGER.exception(exc)
+
+    def _set_genesis(self, block):
+        # This is used by a non-genesis journal when it has recieved the
+        # genesis block from the genesis validator
+        if block.previous_block_id == NULL_BLOCK_IDENTIFIER:
+            chain_id = self._chain_id_manager.get_block_chain_id()
+            if chain_id != block.identifier:
+                LOGGER.warning("Block id does not match block chain id."
```

Comment:
These two sentences would be concatenated with no space between them.  The chain id / block id should be logged as part of the message.